packages/amplify-codegen/src/commands/types.js

```diff
@@ -58,22 +58,31 @@ async function generateTypes(context, forceDownloadSchema, withoutInit = false,
         const target = cfg.amplifyExtension.codeGenTarget;
 
         const excludes = cfg.excludes.map(pattern => `!${pattern}`);
-        const queryFilePaths = glob.sync([...includeFiles, ...excludes], {
+        const queryFilePaths = glob.sync([...includeFiles, ...excludes, `!${generatedFileName}`], {
```

Comment:
Should `generatedFileName` instead be added to the cfg.excludes list during `codegen add/update`? Then I don't think we'd require any other code change, right? Or would we still end up requiring the `Source` filtering down below?

Reply:
Other files that are in the includes glob can lead to the error. i.e. (`src/graphql/foo.ts`). We could say user error at that point, but we should at least improve the error messaging to say to add these files to the excludes.